src/submission/migration/migration-manager.ts

```diff
@@ -310,18 +314,31 @@ export namespace MigrationManager {
   }
 
   function checkValueTypeChanges(
+    diffs: SchemasDictionaryDiffs,
     valueTypeChanges: string[],
     clinicalEntityName: ClinicalEntitySchemaNames,
   ): string[] {
-    const changedFields: string[] = [];
+    const prohibitedChangedFields: string[] = [];
     valueTypeChanges.forEach(change => {
       const changedSchema = change.split('.')[0];
-      if (clinicalEntityName === changedSchema) {
+      if (clinicalEntityName === changedSchema && !allowValueTypeUpdate(diffs, change)) {
         const field: string = change.split('.')[1];
-        changedFields.push(field);
+        prohibitedChangedFields.push(field);
       }
     });
-    return changedFields;
+    return prohibitedChangedFields;
+  }
+
+  function allowValueTypeUpdate(diffs: SchemasDictionaryDiffs, valueTypeChange: string): boolean {
```

Comment:
This function is the main point of the feature, it checks the `diff` of this value type change and verifies `before` and `after` value type to allow change.